backend/src/models/reports/Tab_13_rpt_CF_PortfolioSummary.js

```diff
@@ -1,16 +1,238 @@
+        i.invoice_number, 
+        sum(id.rate * id.unit_amount) amount, 
+        c.Status, 
+        s.supplier_name 
+      from 
+        Contract c 
+        left join invoice i on c.id = i.contract_id 
+        left join fiscal_Year fy on i.fiscal = fy.id 
+        left join invoice_Detail id on i.id = id.invoice_id 
+        left join supplier s on c.supplier_id = s.id 
+      group by 
+        c.id, 
+        c.co_number, 
+        i.fiscal, 
+        fy.fiscal_year, 
+        i.invoice_date, 
+        i.billing_period, 
+        i.invoice_number, 
+        c.status, 
+        s.supplier_name
+    ) AS q1`
+  ),
+
+  q2: knex.raw(
+    `(
+      select 
+        c.ID Contract_ID, 
+        c.CO_Number, 
+        coalesce(t.Fiscal, c.Fiscal) Fiscal, 
+        coalesce(
+          fy.Fiscal_Year, fy_d.Fiscal_Year
+        ) Fiscal_Year, 
+        sic.Portfolio_ID, 
+        po.Portfolio_Name, 
+        po.Portfolio_Abbrev, 
+        c.Start_Date, 
+        p.Project_Number, 
+        c.End_Date, 
+        c.Status, 
+        coalesce(
+          t.Total_Fee_Amount, c.Total_Fee_Amount
+        ) Total_Fee_Amount, 
+        coalesce(
+          t.Total_Expense_Amount, c.Total_Expense_Amount
+        ) Total_Expense_Amount 
+      from 
+        Contract c 
+        inner join Fiscal_Year fy_d on c.Fiscal = fy_d.ID 
+        left join (
+          select 
+            Contract_ID, 
+            Fiscal, 
+            sum(Total_Fee_Amount) Total_Fee_Amount, 
+            sum(Total_Expense_Amount) Total_Expense_Amount 
+          from 
+            (
+              select 
+                Contract_ID, 
+                Fiscal, 
+                sum(Hours * Assignment_Rate) Total_Fee_Amount, 
+                null Total_Expense_Amount 
+              from 
+                Contract_Resource 
+              group by 
+                Contract_ID, 
+                Fiscal 
+              union 
+              select 
+                Contract_ID, 
+                Fiscal, 
+                sum(
+                  case when Is_Expense = False then Deliverable_Amount else cast(0 as money) end
+                ), 
+                sum(
+                  case when Is_Expense = True then Deliverable_Amount else cast(0 as money) end
+                ) 
+              from 
+                Contract_Deliverable 
+              group by 
+                Contract_ID, 
+                Fiscal
+            ) t_sub 
+          group by 
+            Contract_ID, 
+            Fiscal
+        ) t on c.ID = t.Contract_ID 
+        left join Fiscal_Year fy on t.Fiscal = fy.ID 
+        left join Project p on c.Project_ID = p.ID 
+        left join lateral (
+          select 
+            Portfolio_ID 
+          from 
+            SID_Internal_Coding 
+          where 
+            Contract_ID = c.ID 
+          limit 
+            1
+        ) sic on true 
+        left join Portfolio po on po.ID = sic.Portfolio_ID
+    )  AS q2`
+  ),
+};
+
```

Comment:
it's sad that we have these gross queries, but I get the time savings

one thing to look for - I am seeing that knex.js sometimes will timeout and cause a 500 error to be thrown if the query it is wrapping is really complex. 

it might be necessary to re-factor these queries in that case. time will tell